### ISSUE TITLE
Scientific Workflow Provenance

### ISSUE THREAD

**Issue by vsoch** [schemaorg/suggestions-questions-brainstorming#110]
hey @schemaorg!

Thanks for your discussion so far in schemaorg/schemaorg#2059, we are making progress and I hope the meeting with @rvguha in January can solidify some of the work and review discussion with OCI and the toy examples I've created. This is a related issue that I'm working on in parallel - and it hits many more users (including those that don't use containers). The question at hand is:

> How do we use schema.org to represent a scientific workflow?

In the context of containers, if we do have a representation of A Container in schema.org, we would simply define the relationships between them using some kind of Workflow model. Outside of containers, you can have other kinds of SoftwareApplication working together.

## Does schema.org have a model?
From my basic search here, I can't find something that cleanly exists what I'm thinking of. You could say different interactions are kinds of [Actions](https://schema.org/Action) or that a workflow is (sort of) [SoftwareSourceCode](https://schema.org/SoftwareSourceCode) with some kind of Accessibility defined... but that really isn't right. I don't see a clear "scientific workflow" representation that would be desired, to go along with Dataset. Is there initiative working on this?

## Does bioschemas have a model?
I next went back to bioschemas, and I thought I'd find simple concepts for workflows and experiments here. As far as I can tell, bioschemas is working on [biological entties](http://bioschemas.org/specifications/drafts) (akin to genes, proteins, etc.) and less so something like an experiment, a workflow, a step, an input, etc.

## What about ontology?
From @satra I was pointed to what the neuroimaging data model (nidm) community is embracing to describe neuroimaging worklows - [ProvOne](http://vcvcomputing.com/provone/provone.html) is a nicely defined ontology. Has there been discussion about representation of these models into schema.org?

## My Use Case
The Encode Data Portal (Encode-DCC) I beiieve has on the order of hundreds of thousands --> (million?) datasets with comlicated (many non container based) workflows that need to be described. Right now they have a fairly non-standarized description and need something better. In that I'm aware of the sheer number of labs / institutions with this need, and use of schema.org (again) to drive search, and the need to link Datasets with workflows, how can we best go about this?

 - I'm aware there are a million and a half actual workflow managers (e.g., cwl, wdl, or just look at [the awesomepipelines](https://github.com/pditommaso/awesome-pipeline) so we would want to represent a level of abstraction that isn't wed to a particular one, but would generally let any given pipeline export / describe using the standard.

Here are some links to share from master @satra!

 - [Nidm Workflows](https://docs.google.com/presentation/d/10WEaWARvpT05rWFEhOkxq9WCCR1jhFrwuaWQ_ZoYn0k/edit#slide=id.g1fc8e85420_1_0) is a good example of how a specific domain can use ProvOne to describe a niche datatype (brains! :jack_o_lantern: ) So for example, if schema.org can represent ProvOne, that also means there is some kind of mapping to this entire community. Note the slides on use cases and goals - they are very good.
 - [ProvOne Python](https://github.com/incf-nidash/PyNIDM/blob/master/nidm/core/provone.py) not to say that Python is the only rule of the world, but this is to show substantial work has been already done by the nidm team with respect to ProvOne. Python (imho) is the strongest language for scientific programming, as opposed to something like GoLang.

I'll leave it at that to start our discussion! Thanks everyone.

**Comment by RichardWallis** [comment]
See issue #7 for the context of the move from the main Schema.org [issue tracker](https://github.com/schemaorg/schemaorg/issues) to this repository.
